ts/api/api.ts

```diff
@@ -409,45 +440,66 @@ class InitializeOnDemandApi implements OnDemandOpenYoloApi {
     });
   }
 
-  private init(preloadRequest?: PreloadRequest) {
+  private init(timeoutRacer: TimeoutRacer, preloadRequest?: PreloadRequest):
+      Promise<OpenYoloWithTimeoutApi> {
     if (!this.implPromise) {
-      this.implPromise = OpenYoloApiImpl.create(
-          this.providerUrlBase,
-          this.renderMode,
-          this.areTimeoutsDisabled,
-          preloadRequest);
+      this.implPromise = createOpenYoloApi(
+          timeoutRacer, this.providerUrlBase, this.renderMode, preloadRequest);
     }
     return this.implPromise;
   }
 
+  private startTimeoutRacer(defaultTimeoutMs: number): TimeoutRacer {
```

Comment:
Might want to call this method `overrideXXX` because that's what it really does.